exasol-jdbc/src/test/scala/com/exasol/spark/DefaultSourceSuite.scala

```diff
@@ -96,4 +98,44 @@ class DefaultSourceSuite extends AnyFunSuite with Matchers with MockitoSugar {
     // should not contains irrelevant options for exasol
     assert(!newConf.contains("spark.other.options") && !newConf.contains("options"))
   }
+
+  test("`saveDataFrame` should throw exception on null main connection") {
+    val sqlContext = mock[SQLContext]
+    val sparkContext = mock[SparkContext]
+    when(sqlContext.sparkContext).thenReturn(sparkContext)
+    when(sparkContext.defaultParallelism).thenReturn(1)
+    val manager = mock[ExasolConnectionManager]
+    when(manager.writerMainConnection()).thenReturn(null)
+
+    val df = mock[DataFrame]
+    val options = mock[ExasolOptions]
```

Comment:
Moving creation of the mocks and common code like `when(sqlContext.sparkContext).thenReturn(sparkContext)` to setup would make the test shorter.

Reply:
Good point, implemented a helper for `sqlContext` creation